spec/inputs/file_spec.rb

```diff
@@ -190,6 +199,47 @@
     end
   end
 
+  context "when #run is called multiple times", :unix => true do
+    let(:tmpdir_path)  { Stud::Temporary.directory }
+    let(:sincedb_path) { Stud::Temporary.pathname }
+    let(:file_path)    { "#{tmpdir_path}/a.log" }
+    let(:buffer)       { [] }
+    let(:lsof)         { [] }
+    let(:stop_proc) do
+      lambda do |input, arr|
+        Thread.new(input, arr) do |i, a|
+          sleep 0.5
+          a << `lsof -p #{Process.pid} | grep "a.log"`
```

Comment:
We could use [fosl](https://github.com/jordansissel/fosl) for interfacing with lsof instead, but I won't block this PR on that. :)